sdk/digitaltwins/azure-digitaltwins-core/setup.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+# pylint:disable=missing-docstring
+
+import re
+import os.path
+from io import open
+from setuptools import find_packages, setup
+
+# Change the PACKAGE_NAME only to change folder and different name
+PACKAGE_NAME = "azure-digitaltwins-core"
+PACKAGE_PPRINT_NAME = "Azure DigitalTwins Core"
+
+# a-b-c => a/b/c
+PACKAGE_FOLDER_PATH = PACKAGE_NAME.replace("-", "/")
+# a-b-c => a.b.c
+NAMESPACE_NAME = PACKAGE_NAME.replace("-", ".")
+
+# azure v0.x is not compatible with this package
+# azure v0.x used to have a __version__ attribute (newer versions don't)
+try:
+    import azure
+
+    try:
+        VER = azure.__version__  # type: ignore
+        raise Exception(
+            "This package is incompatible with azure=={}. ".format(VER) + 'Uninstall it with "pip uninstall azure".'
+        )
+    except AttributeError:
+        pass
+except ImportError:
+    pass
+
+# Version extraction inspired from 'requests'
+with open(os.path.join(PACKAGE_FOLDER_PATH, "_version.py"), "r") as fd:
+    VERSION = re.search(r'^VERSION\s*=\s*[\'"]([^\'"]*)[\'"]', fd.read(), re.MULTILINE).group(1)
+
+if not VERSION:
+    raise RuntimeError("Cannot find version information")
+
+with open("README.md", encoding="utf-8") as f:
+    README = f.read()
+with open("CHANGELOG.md", encoding="utf-8") as f:
+    CHANGELOG = f.read()
+
+setup(
+    name=PACKAGE_NAME,
+    version=VERSION,
+    description="Microsoft Azure {} Client Library for Python".format(PACKAGE_PPRINT_NAME),
+    long_description=README + "\n\n" + CHANGELOG,
+    long_description_content_type="text/markdown",
+    license="MIT License",
+    author="Microsoft Corporation",
+    author_email="azure-digitaltwins-core@microsoft.com",
+    url="https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/digitaltwins/azure-digitaltwins-core",
+    classifiers=[
+        "Development Status :: 4 - Beta",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 2",
+        "Programming Language :: Python :: 2.7",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.5",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
+        'Programming Language :: Python :: 3.8',
+        "License :: OSI Approved :: MIT License",
+    ],
+    zip_safe=False,
+    packages=find_packages(
+        exclude=[
+            "samples",
+            "tests",
+            # Exclude packages that will be covered by PEP420 or nspkg
+            "azure",
```

Comment:
We will need to add the namespace package from the exclude list as well as the required list - something like this:
```python
    packages=find_packages(exclude=[
        'samples',
        'tests',
        # Exclude packages that will be covered by PEP420 or nspkg
        'azure',
        'azure.digitaltwins',
    ]),
    install_requires=[
        "azure-core<2.0.0,>=1.7.0",
        "msrest>=0.6.10",
    ],
    extras_require={
        ":python_version<'3.0'": ['futures', 'azure-digitaltwins-nspkg'],
        ":python_version<'3.4'": ['enum34>=1.0.4'],
        ":python_version<'3.5'": ["typing"]
    },
)
```

Also I see you have cryptography as a requirement - is this needed? I don't see any encryption etc that would make use of this?

Reply:
You are right the crypto is not needed. Fixed.